Examples/src/main/kotlin/com/mapbox/vision/examples/UsbVideoSourceActivityKt.kt

```diff
@@ -0,0 +1,229 @@
+
+        override fun detach() {
+            backgroundHandler.post {
+                uvcCamera?.stopPreview()
+
+                usbMonitor?.unregister()
+                usbMonitor?.destroy()
+                videoSourceListener = null
+            }
+
+            backgroundHandlerThread.quitSafely()
+        }
+    }
+
+    private var usbMonitor: USBMonitor? = null
+    private var uvcCamera: UVCCamera? = null
+
+    // VisionEventsListener handles events from Vision SDK on background thread.
+    private val visionEventsListener = object : VisionEventsListener {}
+
+    override fun onPermissionsGranted() {
+        startVisionManager()
+    }
+
+    override fun initViews() {
+        setContentView(R.layout.activity_main)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        startVisionManager()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        stopVisionManager()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        vision_view.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        vision_view.onPause()
+    }
+
+    private fun startVisionManager() {
+        if (allPermissionsGranted() && !visionManagerWasInit) {
+            VisionManager.create(usbVideoSource)
+            VisionManager.setModelPerformanceConfig(
+                ModelPerformanceConfig.Merged(
+                    ModelPerformance.On(ModelPerformanceMode.FIXED, ModelPerformanceRate.HIGH)
+                )
+            )
+            vision_view.setVisionManager(VisionManager)
+            VisionManager.visionEventsListener = visionEventsListener
+            VisionManager.start()
+
+            visionManagerWasInit = true
+        }
+    }
+
+    private fun stopVisionManager() {
+        if (visionManagerWasInit) {
+            VisionManager.stop()
+            VisionManager.destroy()
+
+            visionManagerWasInit = false
+        }
+    }
+
+    private val onDeviceConnectListener: USBMonitor.OnDeviceConnectListener =
+        object : USBMonitor.OnDeviceConnectListener {
+            override fun onAttach(device: UsbDevice?) {
+                toast("Usb camera attached")
+                usbMonitor?.requestPermission(device!!)
+            }
+
+            override fun onConnect(
+                device: UsbDevice?,
+                ctrlBlock: USBMonitor.UsbControlBlock?,
+                createNew: Boolean
+            ) {
+                toast("Usb camera connected")
+                backgroundHandler.post {
+                    synchronized(this@UsbVideoSourceActivityKt) {
+                        uvcCamera?.destroy()
+                        uvcCamera = initializeUvcCamera(ctrlBlock!!)
+                    }
+                }
+            }
+
+            override fun onDetach(device: UsbDevice?) {
+                toast("Usb camera detached")
+            }
+
+            override fun onCancel(p0: UsbDevice?) {
+                toast("Usb camera cancelled")
```

Comment:
No action needed? Perhaps use same action as `onDisconnect`.

Reply:
Not necessary, called if connection was not successful and cancelled